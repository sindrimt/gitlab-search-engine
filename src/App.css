@import url("https://fonts.googleapis.com/css?family=Open+Sans|Roboto:400,700&display=swap");
*,
::after,
::before {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    overscroll-behavior-y: none;
}

:root {
    --color-p1: hsl(185, 91%, 17%);
    --color-p2: hsl(185, 84%, 25%);
    --color-p3: hsl(185, 81%, 29%);
    --color-p4: hsl(185, 77%, 34%);
    --color-p5: hsl(185, 62%, 45%);
    --color-p6: hsl(185, 57%, 50%);
    --color-p7: hsl(184, 65%, 59%);
    --color-p8: hsl(184, 80%, 74%);
    --color-p9: hsl(185, 94%, 87%);
    --color-p10: hsl(186, 100%, 94%);
    --color-grey1: hsl(209, 61%, 16%);
    --color-grey2: hsl(211, 39%, 23%);
    --color-grey3: hsl(209, 34%, 30%);
    --color-grey4: hsl(209, 28%, 39%);
    --color-grey5: hsl(210, 22%, 49%);
    --color-grey6: hsl(209, 23%, 60%);
    --color-grey7: hsl(211, 27%, 70%);
    --color-grey8: hsl(210, 31%, 80%);
    --color-grey9: hsl(212, 33%, 89%);
    --color-grey10: #f1f5f8;
    --color-white: #fff;
    --color-red-dark: hsl(360, 67%, 44%);
    --color-red-light: hsl(360, 71%, 66%);
    --color-green-dark: hsl(125, 67%, 44%);
    --color-green-light: hsl(125, 71%, 66%);
    --color-black: #222;
    --font-primary: "Roboto", sans-serif;
    --font-secondary: "Open Sans", sans-serif;
    --transition: all 0.3s linear;
    --spacing: 0.1rem;
    --radius: 0.25rem;
    --light-shadow: 0 5px 15px rgba(0, 0, 0, 0.1);
    --dark-shadow: 0 5px 15px rgba(0, 0, 0, 0.2);
    --max-width: 1170px;
    --fixed-width: 620px;
}

body {
    font-family: "Open Sans", sans-serif;
    background: #f1f5f8;
    color: hsl(209, 61%, 16%);
    line-height: 1.5;
    font-size: 0.875rem;
}

h1,
h2,
h3,
h4 {
    letter-spacing: 0.1rem;
    text-transform: capitalize;
    line-height: 1.25;
    margin-bottom: 0.75rem;
    font-family: "Roboto", sans-serif;
}
h1 {
    font-size: 3rem;
}
h2 {
    font-size: 2rem;
}
h3 {
    font-size: 1.25rem;
}
h4 {
    font-size: 0.875rem;
}
p {
    margin-bottom: 1.25rem;
    color: hsl(210, 22%, 49%);
}

#repoInfo {
    margin: auto;
}

@media screen and (min-width: 800px) {
    h1 {
        font-size: 4rem;
    }
    h2 {
        font-size: 2.5rem;
    }
    h3 {
        font-size: 1.75rem;
    }
    h4 {
        font-size: 1rem;
    }
    body {
        font-size: 1rem;
    }
    h1,
    h2,
    h3,
    h4 {
        line-height: 1;
    }

    
}

.section {
    padding: 1rem 0;
    position: relative;
}

.section-center {
    width: 90vw;
    margin: 0 auto;
    max-width: 1170px;
}
@media screen and (min-width: 992px) {
    .section-center {
        width: 95vw;
    }
}
