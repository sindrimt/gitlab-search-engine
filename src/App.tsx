import React, { useEffect, useContext } from "react";
import "./App.css";

import { StateProvider } from "./context/ContextState";

import Navbar from "./components/Navbar/Navbar";
import Search from "./components/Search/Search";
import Stats from "./components/Stats/Stats";
import Info from "./components/Info/Info";
import Footer from "./components/Footer/Footer";
import { getRepositoryInformation } from "./utils/utils";

const App = () => {
    return (
        <StateProvider>
            <Navbar />
            <Search />
            <Info />
            <Stats />
            <Footer />
        </StateProvider>
    );
};

export default App;
