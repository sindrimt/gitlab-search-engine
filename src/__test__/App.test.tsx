import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

it('App rendered properly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });