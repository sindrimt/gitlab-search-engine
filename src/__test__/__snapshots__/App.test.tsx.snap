// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendered properly 1`] = `
Array [
  <nav
    className="sc-bczRLJ kLzsTs"
  >
    <h4>
      Welcome to 
      <strong>
        GITLAB SEARCH ENGINE
      </strong>
    </h4>
    <img
      alt="Logo"
      className="sc-gsnTZi eUjjzb"
      src="search.png"
    />
  </nav>,
  <section
    className="section"
  >
    <div
      className="sc-hKMtZM hUWbYg section-center"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-control"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <input
            className="searchBox"
            id="input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter Gitlab Repository ID"
            type="text"
          />
          <button
            id="button"
            type="submit"
          >
            search
          </button>
        </div>
      </form>
      <h3
        id="repoInfo"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <section
      className="sc-jSMfEi jdyXAo section-center"
    >
      <article
        className="item"
      >
        <span
          className="red"
        >
          <svg
            className="icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <line
              x1="1.05"
              x2="7"
              y1="12"
              y2="12"
            />
            <line
              x1="17.01"
              x2="22.96"
              y1="12"
              y2="12"
            />
          </svg>
        </span>
        <div>
          <h3>
            <div
              className="sc-gKXOVf SiysH"
            />
          </h3>
          <p>
            commits
          </p>
        </div>
      </article>
      <article
        className="item"
      >
        <span
          className="purple"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M752 100c-61.8 0-112 50.2-112 112 0 47.7 29.9 88.5 72 104.6v27.6L512 601.4 312 344.2v-27.6c42.1-16.1 72-56.9 72-104.6 0-61.8-50.2-112-112-112s-112 50.2-112 112c0 50.6 33.8 93.5 80 107.3v34.4c0 9.7 3.3 19.3 9.3 27L476 672.3v33.6c-44.2 15-76 56.9-76 106.1 0 61.8 50.2 112 112 112s112-50.2 112-112c0-49.2-31.8-91-76-106.1v-33.6l226.7-291.6c6-7.7 9.3-17.3 9.3-27v-34.4c46.2-13.8 80-56.7 80-107.3 0-61.8-50.2-112-112-112zM224 212a48.01 48.01 0 0 1 96 0 48.01 48.01 0 0 1-96 0zm336 600a48.01 48.01 0 0 1-96 0 48.01 48.01 0 0 1 96 0zm192-552a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
            />
          </svg>
        </span>
        <div>
          <h3>
            <div
              className="sc-gKXOVf SiysH"
            />
          </h3>
          <p>
            forks
          </p>
        </div>
      </article>
      <article
        className="item"
      >
        <span
          className="green"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M464 688a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72-112c4.4 0 8-3.6 8-8V296c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48zm400-188h-59.3c-2.6 0-5 1.2-6.5 3.3L763.7 538.1l-49.9-68.8a7.92 7.92 0 0 0-6.5-3.3H648c-6.5 0-10.3 7.4-6.5 12.7l109.2 150.7a16.1 16.1 0 0 0 26 0l165.8-228.7c3.8-5.3 0-12.7-6.5-12.7zm-44 306h-64.2c-5.5 0-10.6 2.9-13.6 7.5a352.2 352.2 0 0 1-49.8 62.2A355.92 355.92 0 0 1 651.1 840a355 355 0 0 1-138.7 27.9c-48.1 0-94.8-9.4-138.7-27.9a355.92 355.92 0 0 1-113.3-76.3A353.06 353.06 0 0 1 184 650.5c-18.6-43.8-28-90.5-28-138.5s9.4-94.7 28-138.5c17.9-42.4 43.6-80.5 76.4-113.2 32.8-32.7 70.9-58.4 113.3-76.3a355 355 0 0 1 138.7-27.9c48.1 0 94.8 9.4 138.7 27.9 42.4 17.9 80.5 43.6 113.3 76.3 19 19 35.6 39.8 49.8 62.2 2.9 4.7 8.1 7.5 13.6 7.5H892c6 0 9.8-6.3 7.2-11.6C828.8 178.5 684.7 82 517.7 80 278.9 77.2 80.5 272.5 80 511.2 79.5 750.1 273.3 944 512.4 944c169.2 0 315.6-97 386.7-238.4A8 8 0 0 0 892 694z"
            />
          </svg>
        </span>
        <div>
          <h3>
            <div
              className="sc-gKXOVf SiysH"
            />
          </h3>
          <p>
            total issues
          </p>
        </div>
      </article>
      <article
        className="item"
      >
        <span
          className="yellow"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
            />
          </svg>
        </span>
        <div>
          <h3>
            <div
              className="sc-gKXOVf SiysH"
            />
          </h3>
          <p>
            stars
          </p>
        </div>
      </article>
    </section>
  </section>,
  <section
    className="section"
  >
    <div
      className="sc-eCYdqJ bqGzMX section-center"
    >
      <div
        id="760aff92-675d-481f-b78a-c8089449eb43"
      />
    </div>
  </section>,
  <section
    className="sc-iBkjds iuidNx"
  >
    @ Made by group 40 - Marcus, Ragnar, Kelvin and Sindri
  </section>,
  <div
    className="sc-ftvSup jVXnOs"
  />,
]
`;
